File: backend/internal/domain/entity/document_types.go
package entity

// Document type constants (Job 07). Kept in sync with the DB check constraint
// in 000007_create_documents_table.up.sql.
const (
	DocTypeProposal          = "proposal"
	DocTypeDraftChapter      = "draft_chapter" // chapter_number: 1-5
	DocTypeSeminarDoc        = "seminar_doc"
	DocTypeDefenseDoc        = "defense_doc"
	DocTypeFinalThesis       = "final_thesis"
	DocTypeRevisionSheet     = "revision_sheet"
	DocTypeEndorsementLetter = "endorsement_letter"
)

// Document status values (Job 07). Kept in sync with the DB check constraint.
const (
	DocStatusPendingReview    = "pending_review"
	DocStatusApproved         = "approved"
	DocStatusRevisionRequired = "revision_required"
)

// ValidDocumentTypes lists every accepted document type.
var ValidDocumentTypes = []string{
	DocTypeProposal,
	DocTypeDraftChapter,
	DocTypeSeminarDoc,
	DocTypeDefenseDoc,
	DocTypeFinalThesis,
	DocTypeRevisionSheet,
	DocTypeEndorsementLetter,
}

// ValidDocumentType reports whether t is a known document type.
func ValidDocumentType(t string) bool {
	for _, v := range ValidDocumentTypes {
		if v == t {
			return true
		}
	}
	return false
}

// SeminarGate lists the documents that must be approved before a student may
// submit a seminar proposal.
var SeminarGate = []string{DocTypeSeminarDoc}

// DefenseGate lists the documents that must be approved before a student may
// submit a defense request.
var DefenseGate = []string{DocTypeDefenseDoc}

File: frontend/src/components/features/documents-page/document-types.ts
import type { DocumentType } from '@/types/document'

// Document types match backend constants (Job 07/21).
export const DOCUMENT_TYPES: { value: DocumentType; label: string }[] = [
  { value: 'proposal', label: 'Proposal' },
  { value: 'draft_chapter', label: 'Draft Bab' },
  { value: 'seminar_doc', label: 'Dokumen Seminar' },
  { value: 'defense_doc', label: 'Dokumen Sidang' },
  { value: 'final_thesis', label: 'Skripsi Final' },
  { value: 'revision_sheet', label: 'Lembar Revisi' },
  { value: 'endorsement_letter', label: 'Surat Pengesahan' },
]
